services/QuillLMS/client/app/bundles/Comprehension/components/studentView/container.tsx

```diff
@@ -374,8 +375,8 @@ export class StudentViewContainer extends React.Component<StudentViewContainerPr
     const { currentActivity, } = activities
     const { submittedResponses, } = session
     if (!currentActivity) return
-
-    const steps =  currentActivity.prompts.map((prompt, i) => {
+    const filteredSteps = _.sortBy( currentActivity.prompts, 'id' );
```

Comment:
Do we have any guarantee that the prompts will have been created in order of ascending order?  It's possible that this is enforced on the front-end, but I don't think there's anything on the back-end that guarantees this.

Reply:
One thought: we could probably sort by `currentActivity.prompts, 'conjunction'` since, by happenstance, 'because', 'but', and 'so' are in alphabetical order.

Reply:
@anathomical I actually noticed that and made that change earlier, not sure why it didn't commit... just pushed again

In regards to the backend, do you have any proposal as to how we could enforce on the backend?

Reply:
I mean, it's possible that we could add a default scope to the model on the back-end so that it always comes out in the intended order, but as long as we have an algorithm that will always work on the front-end (which it looks like we do sorting by conjunction, at least until new conjunctions get added or something which will probably never happen), we should be save.